### PR TITLE
Adds top level exomeseq-gatk3-joint-genotyping.cwl workflow

### LINKED ISSUE #1: User request to workflow stop after initial VCF file
We have had scenarios where the main workflow [exomeseq-gatk3.cwl](https://github.com/bespin-workflows/exomeseq-gatk3/blob/release-4.1/exomeseq-gatk3.cwl) failed in the VariantRecalibrator step. These failures are due to "finnicky-ness" of the software and the input data. User requested the ability to stop early:

>.. we should have your pipeline skip the variant recalibration step and have it end with the generation of the initial VCF file.

This is the last step requested to be run:
https://github.com/bespin-workflows/exomeseq-gatk3/blob/d99cb4cfbb5b78de5e24f14aa40d9e8d96d02825/subworkflows/exomeseq-02-variantdiscovery.cwl#L86-L88

Eventually we expect to be able to run the entire workflow. We already have a subset workflow that does just the preprocessing: [exomeseq-gatk3-preprocessing.cwl](https://github.com/bespin-workflows/exomeseq-gatk3/blob/release-4.1/exomeseq-gatk3-preprocessing.cwl).

Add a new subset workflow that ends at the above specified step.

### PR DESCRIPTION
**Pull request by johnbradley**
Adds `exomeseq-gatk3-joint-genotyping.cwl` workflow that is a subset of the existing [exomeseq-gatk3.cwl](https://github.com/bespin-workflows/exomeseq-gatk3/blob/release-4.1/exomeseq-gatk3.cwl) workflow. The new workflow stops after completing the `joint_genotyping` step in the variantdiscovery subworkflow.

Fixes #1

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [4.2.1] - 2019-04-02
```
Comment (dleehr) [inline CHANGELOG.md]: Wouldn't this be 4.2.0?

Comment (dleehr) [reply CHANGELOG.md]: I know we don't have an established practice yet, but I expected that we'd make code changes first under code review, then update versions/edit changelog/create a release after merging the changes.

File: subworkflows/exomeseq-02-variantdiscovery-joint-genotyping.cwl [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env cwl-runner
```
Comment (dleehr) [inline subworkflows/exomeseq-02-variantdiscovery-joint-genotyping.cwl]: I think we can remove the `-variantdiscovery` from this file name

**Review by dleehr — CHANGES_REQUESTED** [review]
Code looks good. Thanks for being the guinea pig on new development 😀 . We can chat at standup, but I think we need to establish development practices here (like what branch to target and when to create a release, etc)

**Comment by johnbradley** [comment]
Closing this PR in favor of #3 to better model the versioning process

**Pull request closed without merging**